### PR TITLE
Revise gelu

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - add type_prop and backend tests gelu op

### Tickets:
 - 44621

### CODE REVIEW

File: ngraph/test/backend/gelu.in.cpp [hunk 1]
```diff
@@ -89,3 +94,39 @@ NGRAPH_TEST(${BACKEND_NAME}, gelu_f64)
     handle->call_with_validate({result}, {a});
     EXPECT_TRUE(test::all_close_f(input, read_vector<double>(result)));
 }
```
Comment (jdanieck) [inline ngraph/test/backend/gelu.in.cpp:96]: Please rewrite old test cases to also use TestCase class. It'll be much cleaner this way.

File: ngraph/test/type_prop/gelu.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ASSERT_EQ(gelu->get_approximation_mode(), op::GeluApproximationMode::TANH);
+}  
+
+TEST(type_prop, gelu_incompatible_input_type_boolean)
+{
+    auto param = make_shared<op::Parameter>(element::boolean, Shape{1, 32, 32});
+    ASSERT_THROW(std::make_shared<op::v7::Gelu>(param), ngraph::NodeValidationFailure);
+}
+
+TEST(type_prop, gelu_incompatible_input_type_u16)
```
Comment (jdanieck) [inline ngraph/test/type_prop/gelu.cpp:58]: I see in spec that only floating point types are supported. If it is the case, then we should check that also other integer types are throwing.

File: ngraph/test/backend/gelu.in.cpp (hunk 1 above)
Comment (bszmelcz) [reply ngraph/test/backend/gelu.in.cpp:96]: One of the old tests is just gelu_f32 forward pass which I added with TestCase class, the second one is gelu_f64, but from 
what I know f64 is not supported or at least I am getting error telling me that. So should I remove the old test instances?

**Review by jdanieck — APPROVED** [review]

**Comment by ilyachur** [comment]
Please re-run CI

File: ngraph/test/backend/gelu.in.cpp [hunk 3]
```diff
@@ -30,62 +30,84 @@
 #include "util/test_control.hpp"
 #include "util/test_tools.hpp"
 
+#include "util/engine/test_engines.hpp"
```
Comment (ggalieroc) [inline ngraph/test/backend/gelu.in.cpp:33]: Please remove duplication of includes above. Also remove includes related to backend, runtime tensor and util_all_close* which are not needed anymore

File: ngraph/test/runtime/ie/unit_test.manifest [hunk 4]
```diff
@@ -945,8 +945,6 @@ topk_max_sort_index
 topk_min_sort_index
 topk_1d_i32_max_all
 topk_int64
-gelu_f32
-gelu_f64
 gelu_backprop_factor_f32
```
Comment (ggalieroc) [inline ngraph/test/runtime/ie/unit_test.manifest:948]: Are these ones longer needed?

File: ngraph/test/runtime/interpreter/unit_test.manifest [hunk 5]
```diff
@@ -55,7 +55,6 @@ INTERPRETER.min_to_scalar_int8
 INTERPRETER.max_trivial_int8
 INTERPRETER.max_to_scalar_int8
 INTERPRETER.max_3d_to_scalar_double
-INTERPRETER.gelu_f64
 INTERPRETER.gelu_backprop_factor_f64
```
Comment (ggalieroc) [inline ngraph/test/runtime/interpreter/unit_test.manifest:58]: Is this one longer needed?

File: ngraph/test/visitors/op/gelu.cpp [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+ #include "util/visitor.hpp"
+
+ using namespace std;
+ using namespace ngraph;
+ using ngraph::test::NodeBuilder;
+
+ TEST(attributes, gelu_op)
+ {
+	NodeBuilder::get_ops().register_factory<opset7::Gelu>();
```
Comment (ggalieroc) [inline ngraph/test/visitors/op/gelu.cpp:19]: nitpick: indentation misalignment

File: ngraph/test/runtime/ie/unit_test.manifest (hunk 4 above)
Comment (bszmelcz) [reply ngraph/test/runtime/ie/unit_test.manifest:948]: I have a discussion with @jdanieck and we came to the conclusion that they are not needed anymore.

Comment (ggalieroc) [reply ngraph/test/runtime/ie/unit_test.manifest:948]: I am referring to `gelu_backprop_factor_f32`, if not maybe we can remove it? Same as with `INTERPRETER.gelu_backprop_factor_f64`

Comment (bszmelcz) [reply ngraph/test/runtime/ie/unit_test.manifest:948]: oh ok, I think we can remoe it

**Review by ggalieroc — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**